analyses/chromosomal-instability/02b-plot-chr-instability-by-histology.Rmd

```diff
@@ -0,0 +1,165 @@
+`%>%` <- dplyr::`%>%`
+
+# Need the plotting functions
+source(file.path("util", "chr-break-plot.R"))
+```
+
+Set up option for how many samples minimum we need to include in CDF plot. 
+Borrowed this type of logic from @sjspielman in the [gene-set-enrichment analysis.](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/ee1283991d30237a6a04be8fc6fec7672844133b/analyses/gene-set-enrichment-analysis/02-model-gsea.Rmd#L55)
+
+```{r}
+# Let's check if this is numeric
+if (!(is.numeric(params$min_samples))) {
+  warning("The parameter min_samples should be numeric. Trying to coerce to numeric.")
+  params$min_samples <- as.numeric(params$min_samples)
+} 
+```
+
+### Directories and Files
+
+```{r}
+# Path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# Path to output directory
+plots_dir <- "plots"
+
+# Path to tumor type plots output directory
+hist_plots_dir <- file.path(plots_dir, "tumor-type")
+
+# Create the hist_plots_dir  if it does not exist
+if (!dir.exists(hist_plots_dir)) {
+  dir.create(hist_plots_dir, recursive = TRUE)
+}
+```
+
+Here's all the input files we will need: 
+
+```{r}
+metadata_file <- file.path(data_dir, "pbta-histologies.tsv")
+hist_binned_breaks_file <- file.path("breakpoint-data", "histology_breakpoint_binned_counts.RDS")
+intersection_densities_file <- file.path("breakpoint-data", "intersection_of_breaks_densities.tsv")
+cnv_densities_file <- file.path("breakpoint-data", "cnv_breaks_densities.tsv")
+sv_densities_file <- file.path("breakpoint-data", "sv_breaks_densities.tsv")
+```
+
+Output plots: 
+
+```{r}
+intersection_cdf_plot_file <- file.path(plots_dir, "intersection_breaks_cdf_plot.png")
+cnv_cdf_plot_file <- file.path(plots_dir, "cnv_breaks_cdf_plot.png")
+sv_cdf_plot_file <- file.path(plots_dir, "sv_breaks_cdf_plot.png")
+```
+
+### Read in data 
+
+Set up metadata
+
+```{r}
+# Read in the metadata
+metadata <- readr::read_tsv(metadata_file, guess_max = 10000)
+```
+
+Load in the localized by histology breakpoint data. 
+
+```{r}
+hist_break_local <- readr::read_rds(hist_binned_breaks_file)
+```
+
+## Plot CDF plot by histology
+
+#### Intersection breaks density plot. 
+
+```{r}
+breaks_cdf_plot(density_file = intersection_densities_file, 
+                metadata_df = metadata, 
+                cdf_plot_file = intersection_cdf_plot_file)
+```
+
+#### CNV breaks density plot. 
+
+```{r}
+breaks_cdf_plot(density_file = cnv_densities_file, 
+                metadata_df = metadata, 
+                cdf_plot_file = cnv_cdf_plot_file)
+```
+
+#### SV breaks density plot. 
+
+```{r}
+breaks_cdf_plot(density_file = sv_densities_file, 
+                metadata_df = metadata, 
+                cdf_plot_file = sv_cdf_plot_file)
+```
+
+### Plot the breakpoints for each tumor-type
+
+Here we will plot median number of break points for the tumor-type group per 
+each bin.
+
+```{r}
+purrr::imap(hist_break_local, function(.x, name = .y) {
+  # Make the combo plot
+  multipanel_break_plot(
```

Comment:
When I run this locally, I get a big list of NULL outputs, which isn't a problem, but isn't so pretty. My quick suggestion would be have `multipanel_break_plot` return something (the plot object?), and then have the call to `purrr::imap()` saved to something new.

Or, change the block to hide output with results=FALSE, though for some reason this is giving trouble

Reply:
I added a `results=FALSE` so you don't have to see this hideous output. :)